src/Ledger/GovernanceActions.lagda

```diff
@@ -113,6 +171,30 @@ NeedsHash (ChangePParams _ _)  = GovActionID
 NeedsHash (TreasuryWdrl _)     = ⊤
 NeedsHash Info                 = ⊤
 
+HashProtected : Set → Set
+HashProtected A = A × GovActionID
+\end{code}
+} %% end small
+\caption{NeedsHash and HashProtected types}
+\label{fig:needshash-and-hashprotected-types}
+\end{figure*}
+Figure~\ref{fig:needshash-and-hashprotected-types} defines types that are used in
+ratification (for verifyPrev) where we check that the stored hash matches the one
+attached to the action we want to ratify.
+\begin{itemize}
+\item
+  \defn{Ratification}. An action is said to be \defn{ratified} when it gathers enough votes in its favor
+  (according to the rules described in Section~\ref{sec:ratification}).
+\item
+  \defn{Expiration}. An action that doesn't collect sufficient `yes' votes before its deadline is said to have \defn{expired}.
```

Comment:
Okay, I'll change it back.  I had changed all yes, no and abstains from the Agda colored font to just quoted words, since I got the impression from your earlier comments that we want to make the docs more prosey and less codey... anyway, I'll change them all (or at least some of them) back.